cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -221,75 +278,72 @@ private void processMetrics() {
         MessageFetcher fetcher = messagingService.prepareFetch(topicIdMetaKey.getTopicId());
         fetcher.setLimit(fetcherLimit);
         if (lastMessageId != null) {
+          LOG.debug("Start fetching from lastMessageId = {}", lastMessageId);
```

Comment:
`lastMessageId` is a `byte[]`. To log it properly, you need to call `Bytes.toStringBinary(lastMessageId)`. Also, to avoid the string formation unconditionally, better have a `if (LOG.isDebugEnabled())` to guard it.